server/isomorphic-routing/index.js

```diff
@@ -96,10 +102,18 @@ function compose( ...functions ) {
 }
 
 export function getCacheKey( context ) {
+	// express.js Request object has a path property for the pathname (path without the querystring)
```

Comment:
Maybe it's just me, but I'm super confused by this function.

I'm not sure what we receive here as an argument,
I mean, we call it here `context`, but:

[here we pass](https://github.com/Automattic/wp-calypso/blob/5c1ba6433c69a6a35aec23a588f42344031e511c/server/pages/index.js#L146) express request
[here we pass](https://github.com/Automattic/wp-calypso/blob/2e0452a50b013a60642e3f8abf799f4587466f61/server/render/index.js#L85) context

IMO the argument named `context` here needs a different naming, maybe even change signature.

Also, if we can path either a request or either request how does this work when we pass request:
```
if ( isEmpty( context.query ) || isEmpty( context.cacheQueryKeys ) ) { ... }
```

Is that supposed to detect whether that's a context object or a request object?

TL;DR:
* We should make super clear what this function can receive from it's signature / jsdoc
* If we're trying to detect what is actually received ( context or request ) lets say why we're doing it ( we don't cache that? we cache it? we ignore it? )